File: codes/yujo/739_dailytemperatures.ts
function dailyTemperatures(temperatures: number[]): number[] {
  const result: number[] = [];

  for (let i = 0; i < temperatures.length; i++) {
    let days = 0;
    for (let j = i + 1; j < temperatures.length; j++) {
      if (temperatures[j] > temperatures[i]) {
        days = j - i;
        break;
      }
    }
    result.push(days);
  }

  return result;
}

File: codes/hanjun/739_dailytemperatures.py
def dailyTemperatures(self, temperatures:List[int])->List[int]:
    stack = []
    answer = [0]*len(temperature)
    for i, temperature in enumerate(temperatures):
        while stack and temperatures[stack[-1]] < temperature:
            day = stack.pop()
            answer[day] = i - day
		stack.append(i)
        
    return answer